contrib/IECoreAlembic/src/IECoreAlembic/AlembicScene.cpp

```diff
@@ -267,6 +319,493 @@ class AlembicScene::AlembicReader : public AlembicIO
+							tmpValue[12],
+							tmpValue[13],
+							tmpValue[14],
+							tmpValue[15]
+						)
+					);
+				}
+			}
+			else if( pod == kFloat64POD )
+			{
+				if( extent == 1 )
+				{
+					double value;
+					scalarPropertyReader->getSample( sampleIndex, &value );
+					return new IECore::DoubleData( value );
+				}
+				else if( extent == 2 )
+				{
+					double tmpValue[2];
+					scalarPropertyReader->getSample( sampleIndex, &tmpValue[0] );
+					return new IECore::V2dData( V2d( tmpValue[0], tmpValue[1] ), cortexInterpretation );
+				}
+				else if( extent == 3 )
+				{
+					double tmpValue[3];
+					scalarPropertyReader->getSample( sampleIndex, &tmpValue[0] );
+					return new IECore::V3dData( V3d( tmpValue[0], tmpValue[1], tmpValue[2] ), cortexInterpretation );
+				}
+				else if( extent == 4 )
+				{
+					double tmpValue[4];
+					scalarPropertyReader->getSample( sampleIndex, &tmpValue[0] );
+					if( interpretation == "quat" )
+					{
+						return new IECore::QuatdData( Quatd( tmpValue[0], tmpValue[1], tmpValue[2], tmpValue[3] ) );
+					}
+					else if( interpretation == "box" )
+					{
+						return new IECore::Box2dData( Box2d( V2d( tmpValue[0], tmpValue[1] ), V2d( tmpValue[2], tmpValue[3] ) ) );
+					}
+				}
+				else if( extent == 6 )
+				{
+					double tmpValue[6];
+					scalarPropertyReader->getSample( sampleIndex, &tmpValue[0] );
+					return new IECore::Box3dData( Box3d( V3d( tmpValue[0], tmpValue[1], tmpValue[2] ), V3d( tmpValue[3], tmpValue[4], tmpValue[5] ) ) );
+				}
+				else if( extent == 9 )
+				{
+					double tmpValue[9];
+					scalarPropertyReader->getSample( sampleIndex, &tmpValue[0] );
+					return new IECore::M33dData(
+						M33d(
+							tmpValue[0], tmpValue[1], tmpValue[2], tmpValue[3], tmpValue[4], tmpValue[5], tmpValue[6], tmpValue[7], tmpValue[8]
+						)
+					);
+				}
+				else if( extent == 16 )
+				{
+					double tmpValue[16];
+					scalarPropertyReader->getSample( sampleIndex, &tmpValue[0] );
+					return new IECore::M44dData(
+						M44d(
+							tmpValue[0],
+							tmpValue[1],
+							tmpValue[2],
+							tmpValue[3],
+							tmpValue[4],
+							tmpValue[5],
+							tmpValue[6],
+							tmpValue[7],
+							tmpValue[8],
+							tmpValue[9],
+							tmpValue[10],
+							tmpValue[11],
+							tmpValue[12],
+							tmpValue[13],
+							tmpValue[14],
+							tmpValue[15]
+						)
+					);
+				}
+			}
+			else if( pod == kStringPOD )
+			{
+				std::string tmpStr;
+				scalarPropertyReader->getSample( sampleIndex, &tmpStr );
+
+				return new IECore::StringData( tmpStr );
+			}
+
+			IECore::msg(
+				IECore::Msg::Warning,
+				"AlembicScene::readAttributeAtSample",
+				boost::format( "Unsupported attribute type datatype: \"%1%\" extend:%2% interpretation:\"%3%\"" ) % pod % extent % interpretation
+			);
+
+			return nullptr;
+		}
+
+		void attributeHash( double time, IECore::MurmurHash &h ) const
+		{
+			if( !m_xform )
+			{
+				return;
+			}
+
+			h.append( fileName() );
```

Comment:
Can we avoid hashing these in when there are no user properties? It seems that could be quite a common case, and it'd be well worth it in terms of downstream performance.